src/components/search-by-concepts/search-by-concepts.resource.ts

```diff
@@ -1,47 +1,42 @@
-import { openmrsFetch, fhir } from "@openmrs/esm-framework";
+import { openmrsFetch, getGlobalStore } from "@openmrs/esm-framework";
 
-/**
- * This is a somewhat silly resource function. It searches for a patient
- * using the REST API, and then immediately gets the data using the FHIR
- * API for the first patient found. OpenMRS API endpoints are generally
- * hit using `openmrsFetch`. For FHIR endpoints we use the FHIR API
- * object.
- *
- * See the `fhir` object API docs: https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-api/docs/API.md#fhir
- * See the docs for the underlying fhir.js Client object: https://github.com/FHIR/fhir.js#api
- * See the OpenMRS FHIR Module docs: https://wiki.openmrs.org/display/projects/OpenMRS+FHIR+Module
- * See the OpenMRS REST API docs: https://rest.openmrs.org/#openmrs-rest-api
- *
- * @param query A patient name or ID
- * @returns The first matching patient
- */
-export async function getPatient(query) {
-  const searchResult = await openmrsFetch(
-    `/ws/rest/v1/patient?q=${query}&limit=1`,
-    {
-      method: "GET",
-    }
-  );
-  return (
-    await fhir.read<fhir.Patient>({
-      type: "Patient",
-      patient: searchResult.data.results[0].uuid,
-    })
-  ).data;
-}
+import { addToHistory } from "../../cohort-builder.utils";
 
-export async function getConcepts(conceptName: String) {
-  const searchResult = await openmrsFetch(
-    `/ws/rest/v1/concept?v=full&q=${conceptName}`,
-    {
-      method: "GET",
-    }
-  );
+const patientsStore = getGlobalStore("patients");
+const notificationStore = getGlobalStore("notification");
 
-  return (
-    await fhir.read<fhir.ConceptMap>({
-      type: "Patient",
-      patient: searchResult.data.results[0].uuid,
+/**
+ * @param queryDetails query details
+ * @param description query description
+ */
+export const search = async (queryDetails, description: string) => {
+  await openmrsFetch("/ws/rest/v1/reportingrest/adhocquery?v=full", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: queryDetails.query,
+  })
+    .then((response) => {
+      const { data } = response;
+      if (data.error) {
+        notificationStore.setState({
+          notification: { kind: "error", title: data.error },
+        });
+      } else {
+        data.searchDescription = description || queryDetails.label;
+        data.query = queryDetails.query;
+        patientsStore.setState({ patients: data.rows });
+        notificationStore.setState({
+          notification: {
+            kind: "success",
+            title: `Search completed with ${data.rows.length} result(s)`,
+          },
+        });
+        addToHistory(description, data.rows, queryDetails.query);
```

Comment:
This one as well. We can use this function to just do a search and return a patient list.